vcpkg's OpenSSL port exposes OpenSSL::applink rather than OpenSSL::Crypto
for the checks and link step used here.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dde136b..6365a37 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -104,13 +104,13 @@ if (USE_CRYPTO STREQUAL "OpenSSL")
 	# functionality.
 	cmake_push_check_state()
 		set(CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
-		set(CMAKE_REQUIRED_LIBRARIES OpenSSL::Crypto)
+		set(CMAKE_REQUIRED_LIBRARIES OpenSSL::applink)
 		if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
 			list(APPEND CMAKE_REQUIRED_LIBRARIES ws2_32 crypt32)
 		endif()
 	cmake_pop_check_state()
 	cmake_push_check_state()
-		set(CMAKE_REQUIRED_LIBRARIES OpenSSL::Crypto)
+		set(CMAKE_REQUIRED_LIBRARIES OpenSSL::applink)
 		if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
 			list(APPEND CMAKE_REQUIRED_LIBRARIES ws2_32 crypt32)
 		endif()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ceca7fc..7d05082 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -204,7 +204,7 @@ endfunction()
 function(gns_set_target_crypto_properties TGT)
 	if(USE_CRYPTO STREQUAL "OpenSSL" OR USE_CRYPTO25519 STREQUAL "OpenSSL")
 		target_link_libraries(${TGT} PUBLIC
-			OpenSSL::Crypto
+			OpenSSL::applink
 			)
 		if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
 			target_link_libraries(${TGT} PUBLIC
